frontend/src/components/DocumentViewer/DocumentViewBox.tsx

```diff
@@ -1,36 +1,25 @@
-import DocViewer, { DocViewerRenderers } from '@cyntler/react-doc-viewer';
-import { useEffect, useState } from 'react';
+import DocViewer, {
+	TXTRenderer,
+	CSVRenderer,
+	IDocument,
+} from '@cyntler/react-doc-viewer';
+import { useState } from 'react';
+import { ThreeDots } from 'react-loader-spinner';
 
 import OverlayHeader from '@src/components/Overlay/OverlayHeader';
-import { DocumentMeta } from '@src/models/document';
-import { documentService } from '@src/service';
 
 import DocumentViewBoxNav from './DocumentViewBoxNav';
 
 import './DocumentViewBox.css';
 
-const emptyList: DocumentMeta[] = [];
-
-function DocumentViewBox({ closeOverlay }: { closeOverlay: () => void }) {
-	const [documentMetas, setDocumentMetas] = useState<DocumentMeta[]>(emptyList);
+function DocumentViewBox({
+	documents,
+	closeOverlay,
+}: {
+	documents?: IDocument[];
+	closeOverlay: () => void;
+}) {
 	const [documentIndex, setDocumentIndex] = useState<number>(0);
```

Comment:
Interesting story: I tried moving this "current document index" into the parent component, so that the viewer would remember where you were when reopening. Didn't work.

Why? Because even though the JSX that declares our DocumentViewBox belongs to the parent (MainComponent.tsx), we pass that to Overlay component which is rendered by the grandparent (App.tsx). We can store the state in the parent, and when calling setDocumentIndex the parent is indeed updated, and in theory DocumentViewBox will receive the new prop. But there is nothing to trigger App component to update: its state does not change, it has no props, and it is not using context. So Overlay doesn't update either, meaning it continues to render its now outdated "children" with the old prop values.

We cannot simply lift the state and the DocumentViewBox JSX into App, because the documents are fetched in the level/start request which is made by MainComponent. This kind of limitation has come up before - see `mainBodyKey` state in MainComponent - and is another example of how storing level state in MainComponent with a useState hook, and passing it all down via prop drilling, is making life uncomfortable. We might be better off switching to a context-based store instead, for level state.

Reply:
Always teaching!
well explained